app/models/ontology/sentences.rb

```diff
@@ -1,135 +1,151 @@
-      translated = TranslatedSentence.where(audience_id: self)
-      sentence_ids = translated.pluck(:sentence_id)
-      imported = self.direct_imported_ontologies.reduce([[], []]) do |arr, ontology|
-        other_split = ontology.split_translated_sentences
-        other_translated = other_split.first
-        other_translated.delete_if do |translated_sentence|
-          sentence_ids.include?(translated_sentence.sentence_id)
-        end
-        other_sentences = other_split.last
-        other_sentences.delete_if do |sentence|
-          sentence_ids.include?(sentence.id)
-        end
-        other_translated.each { |ot| arr.first << ot }
-        other_sentences.each { |os| arr.last << os }
-        arr
+      def split_translated_sentences
+        translated = TranslatedSentence.where(audience_id: self)
+        sentence_ids = translated.pluck(:sentence_id)
+        imported =
+          self.direct_imported_ontologies.reduce([[], []]) do |arr, ontology|
+            other_split = ontology.split_translated_sentences
+            other_translated = other_split.first
+            other_translated.delete_if do |translated_sentence|
+              sentence_ids.include?(translated_sentence.sentence_id)
+            end
+            other_sentences = other_split.last
+            other_sentences.delete_if do |sentence|
+              sentence_ids.include?(sentence.id)
+            end
+            other_translated.each { |ot| arr.first << ot }
+            other_sentences.each { |os| arr.last << os }
+            arr
+          end
+        [translated + imported.first, self.sentences + imported.last]
       end
-      [translated + imported.first, self.sentences + imported.last]
-    end
 
-    # Find import-links which describe the following mapping:
-    # some ontology imports self.
-    def incoming_imports_with_mappings
-      # INNER JOINS always return an empty result set if the
-      # ON clause does not match.
-      Link.joins(:entity_mappings).where(source_id: self, kind: 'import')
-    end
+      # Find import-mappings which describe the following mapping:
+      # some ontology imports self.
+      def incoming_imports_with_mappings
+        # INNER JOINS always return an empty result set if the
+        # ON clause does not match.
+        Mapping.joins(:symbol_mappings).where(source_id: self, kind: 'import')
+      end
 
-    def create_translated_sentences
-      query, args = links_by_kind_query(self, 'import')
-      imported_ontology_ids = pluck_select([query, *args], :ontology_id).reverse
-      imported_ontology_ids.each { |o_id| Ontology.find(o_id).create_translated_sentences }
-      import_links = incoming_imports_with_mappings
-      if import_links.any?
-        combined_sentences.each do |sentence|
-          import_links.each do |import_link|
-            import_link.entity_mappings.each do |mapping|
-              translate_sentence(sentence, mapping)
+      def create_translated_sentences
```

Comment:
I don't see either, at least not in this branch (maybe later as part of a larger refactoring).